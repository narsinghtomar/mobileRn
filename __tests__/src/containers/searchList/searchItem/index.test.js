/**
 * Button
 */
import 'react-native';
import React from 'react';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';
import SearchItem from '../../../../../src/containers/searchList/searchItem';

describe('SearchItem', () => {
  const searchArticlesReducer = {
    isFatching: true,
    searchHistoryList: [],
  };

  const initialState = {
    searchArticlesReducer,
  };

  const mockStore = configureStore();
  let store;

  it('SearchItem renders correctly', async () => {
    jest.useFakeTimers();
    store = mockStore(initialState);
    const tree = await renderer
      .create(
        <Provider store={store}>
          <SearchItem searchHistoryList={[]} />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
