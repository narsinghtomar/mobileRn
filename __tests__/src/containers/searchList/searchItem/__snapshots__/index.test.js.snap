// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchItem SearchItem renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "borderColor": "#C4C4C4",
      "borderRadius": 18,
      "borderWidth": 0.5,
      "flexDirection": "column",
      "justifyContent": "center",
      "marginVertical": 4,
      "paddingHorizontal": 15,
      "paddingVertical": 10,
      "width": "100%",
    }
  }
>
  <Text
    ellipsizeMode="tail"
    numberOfLines={2}
    style={
      Object {
        "color": "#7C7E83",
        "fontSize": 14,
        "fontWeight": "500",
        "textAlign": "left",
      }
    }
  />
</View>
`;
